### ISSUE TITLE
Bug in conditional statement linting

### ISSUE THREAD

**Issue by iamdarrenhall** [brigade/scss-lint#409]
Hi, I'm getting a linting bug with the following mixin:  

```
@mixin svg($name, $pseudo: false, $display: inline-block) {
    @if($pseudo) {
        &:#{$pseudo} {
            @include svg--content($name, $display);
        }
    } @else {
        @include svg--content($name, $display);
    }
}
```
The problem seems to be with the conditional statement, as when I change it to the following it lints without any issues.

```
@mixin svg($name, $pseudo: false, $display: inline-block) {
    @if($pseudo) {
        &:#{$pseudo} {
            @include svg--content($name, $display);
        }
    }

    @if($pseudo==false) {
        @include svg--content($name, $display);
    }
}
```

Here's the output from scss-lint, cheers! 

```
SCSSLint::Linter::Indentation raised unexpected error linting file /repos/[directory]/source/sass/_sprite.scss: 'undefined method `line' for nil:NilClass'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter/indentation.rb:194:in `node_indent'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter/indentation.rb:184:in `one_shift_greater_than_parent?'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter/indentation.rb:159:in `check_arbitrary_indent'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter/indentation.rb:53:in `check_indent_width'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter/indentation.rb:39:in `check_indentation'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter/indentation.rb:17:in `check_and_visit_children'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter/indentation.rb:61:in `visit_if'
/Library/Ruby/Gems/2.0.0/gems/sass-3.4.3/lib/sass/tree/visitors/base.rb:36:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter.rb:127:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter/indentation.rb:62:in `visit_if'
/Library/Ruby/Gems/2.0.0/gems/sass-3.4.3/lib/sass/tree/visitors/base.rb:36:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter.rb:127:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter.rb:138:in `block in visit_children'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter.rb:136:in `each'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter.rb:136:in `visit_children'
/Library/Ruby/Gems/2.0.0/gems/sass-3.4.3/lib/sass/tree/visitors/base.rb:36:in `block in visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter/indentation.rb:20:in `check_and_visit_children'
/Library/Ruby/Gems/2.0.0/gems/sass-3.4.3/lib/sass/tree/visitors/base.rb:36:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter.rb:127:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter.rb:138:in `block in visit_children'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter.rb:136:in `each'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter.rb:136:in `visit_children'
/Library/Ruby/Gems/2.0.0/gems/sass-3.4.3/lib/sass/tree/visitors/base.rb:36:in `block in visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter/indentation.rb:10:in `visit_root'
/Library/Ruby/Gems/2.0.0/gems/sass-3.4.3/lib/sass/tree/visitors/base.rb:36:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter.rb:127:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/linter.rb:21:in `run'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/runner.rb:56:in `run_linter'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/runner.rb:37:in `block in find_lints'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/runner.rb:35:in `each'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/runner.rb:35:in `find_lints'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/runner.rb:21:in `block in run'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/runner.rb:20:in `each'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/runner.rb:20:in `run'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/cli.rb:51:in `scan_for_lints'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/cli.rb:45:in `act_on_options'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/lib/scss_lint/cli.rb:24:in `run'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.35.0/bin/scss-lint:6:in `<top (required)>'
/usr/bin/scss-lint:23:in `load'
/usr/bin/scss-lint:23:in `<main>'
```

**Comment by sds** [comment]
Hey @iamdarrenhall, thanks for filing a report.

I'm unable to reproduce this. When I copy/paste your example into a file and run `scss-lint`, it reports no lints (and doesn't dump a stack trace). I've tried this both with version 0.35.0 of `scss-lint` and the latest master, and with the `sass` gem explicitly pinned to the version you list (3.4.3).

Can you verify that pasting the code you provided reproduces this issue on your end? (there might be some weird control characters in your file that are lost in translation)

If it's still failing, can you try updating to the latest version of the `sass` gem (3.4.13 at time of writing) and see if that fixes anything?

**Comment by sds** [comment]
Closing due to lack of response.

**Comment by iamdarrenhall** [comment]
Sorry about the delay in responding. Further investigation has revealed that the bug is only being caused when the indentation linter is being used as follows: 

```
linters:

    Indentation:
        allow_non_nested_indentation: true
```

I'm currently using sass 3.4.13 and scss-lint 0.38.0.

**Comment by iamdarrenhall** [comment]
I know next to no ruby, but having a mess around with `/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.38.0/lib/scss_lint/linter.rb` it looks like an IfNode which is an Else doesn't get assigned a node_parent value. 

I added a couple of logs in the function at line 183:

```
def one_shift_greater_than_parent?(node, actual_indent)
  if node.is_a?(Sass::Tree::IfNode)
    puts node
    puts node.node_parent
  end

  parent_indent = node_indent(node.node_parent).length
  expected_indent = parent_indent + @indent_width
  expected_indent == actual_indent
end
```

Which gives me

```
#<Sass::Tree::IfNode:0x007fff32359fd8>
#<Sass::Tree::MixinDefNode:0x007fff3235b428>
#<Sass::Tree::IfNode:0x007fff32362110>
```

So the first IfNode has a parent of MixinDefNode, but the second IfNode (the Else) doesn't have a parent. 

I think that's about as much help as I can be without learning ruby, hope it's useful!

**Comment by sds** [comment]
Thanks @iamdarrenhall, that information proved useful.

I've fixed this in 28e42c1.

**Comment by iamdarrenhall** [comment]
Glad I could help. Thanks for fixing this and all the other hard work you do on the linter, it's much appreciated!

**Comment by szarouski** [comment]
Hi @sds, the issue is happening for me with sass-lint -v 0.38.0 for following code:
```scss
@for $i from 0 through 2 {
    @if ($i) {
    } @else {
    }
}
```
and doesn't happen for following:
```scss
@for $i from 0 through 2 {
    @if ($i) {
    }
}
```

stack trace
```
SCSS Lint
SCSSLint::Linter::Indentation raised unexpected error linting file live-style-guide\specs\RewardsCentre\Shared\grids\temp.scss: 
'undefined method `line' for nil:NilClass' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter/indentation.rb:194
:in `node_indent' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter/indentation.rb:184
:in `one_shift_greater_than_parent?' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter/indentation.rb:159
:in `check_arbitrary_indent' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter/indentation.rb:53
:in `check_indent_width' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter/indentation.rb:39
:in `check_indentation' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter/indentation.rb:17
:in `check_and_visit_children' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter/indentation.rb:61
:in `visit_if' c:/Ruby2/lib/ruby/gems/2.2.0/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:36
:in `visit' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter.rb:131
:in `visit' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter/indentation.rb:62
:in `visit_if' c:/Ruby2/lib/ruby/gems/2.2.0/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:36
:in `visit' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter.rb:131
:in `visit' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter.rb:142
:in `block in visit_children' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter.rb:140
:in `each' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter.rb:140
:in `visit_children' c:/Ruby2/lib/ruby/gems/2.2.0/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:36
:in `block in visit' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter/indentation.rb:20
:in `check_and_visit_children' c:/Ruby2/lib/ruby/gems/2.2.0/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:36
:in `visit' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter.rb:131
:in `visit' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter.rb:142
:in `block in visit_children' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter.rb:140
:in `each' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter.rb:140
:in `visit_children' c:/Ruby2/lib/ruby/gems/2.2.0/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:36
:in `block in visit' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter/indentation.rb:10
:in `visit_root' c:/Ruby2/lib/ruby/gems/2.2.0/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:36
:in `visit' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter.rb:131
:in `visit' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/linter.rb:25
:in `run' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/runner.rb:48
:in `run_linter' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/runner.rb:29
:in `block in find_lints' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/runner.rb:27
:in `each' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/runner.rb:27
:in `find_lints' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/runner.rb:17
:in `block in run' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/runner.rb:16
:in `each' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/runner.rb:16
:in `run' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/cli.rb:54
:in `scan_for_lints' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/cli.rb:48
:in `act_on_options' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/lib/scss_lint/cli.rb:27
:in `run' c:/Ruby2/lib/ruby/gems/2.2.0/gems/scss-lint-0.38.0/bin/scss-lint:6
:in ` ' c:/Ruby2/bin/scss-lint:23
:in `load' c:/Ruby2/bin/scss-lint:23
:in ` ' Report this bug at https://github.com/brigade/scss-lint/issue
```

thank you

**Comment by sds** [comment]
Hey @szarouski, as mentioned above this issue has been fixed on `master`, but has not been released. I'll try to put out a release by the end of this week.

**Comment by szarouski** [comment]
Oh, sorry, I just looked at the commit above and in changelog.md it was saying 0.38.0. My bad.
